### DISCUSSION TITLE
Commute time computation

### DISCUSSION THREAD

**Discussion by Xiaohui9607** [networkit/networkit#976] [Q&A]
Hi, I am trying to compute the commute time and convert it into the effective resistance. The results I get from the networkit does not match the one I got in networkx, here is a demo:

```
b = nx.Graph()
b.add_nodes_from([0,1,2,3])
b.add_edges_from([(0,1),(1,2),(2,3)])
b_nk = nk.nxadapter.nx2nk(b)
b_nk.indexEdges()
ctd = nk.distance.CommuteTimeDistance(b_nk, tol=0.01)
ctd.run()
reff = nx.resistance_distance(b, 0, 3)
ct = ctd.distance(0,3)
```
I got reff = 3, and ct = 4.2426, we know that C(0,3) = 2mR(0,3), so C(s,t) should be 18 instead of 4.2426. Am I missing something?

**Comment by fabratu** [comment]
Hi,

NetworKit computes the Euclidean Commute Time Distance (ECTD), which is the embedding of the graph / commute time in euclidean space. The difference in the result is a `sqrt`, so your relation transforms to C(u,v)^2 = 2mR(u,v).

**Comment by fabratu** [comment]
I will convert this issue to a discussion since it might be valuable for others.
